src/v/cluster/scheduling/partition_allocator.h

```diff
@@ -104,5 +112,10 @@ class partition_allocator {
 
     std::unique_ptr<allocation_state> _state;
     allocation_strategy _allocation_strategy;
+    ss::sharded<members_table>& _members;
+
+    config::binding<std::optional<size_t>> _memory_per_partition;
+    config::binding<std::optional<int32_t>> _fds_per_partition;
+    config::binding<size_t> _fallocation_step;
```

Comment:
We will probably want a "space manager" at some point, which could take the place of passing these separate arguments.